object_detection/Mask_RCNN/mask_rcnn_run_loop.py

```diff
@@ -0,0 +1,181 @@
+
+  This module repeatedly runs 1 training epoch and then evaluation
+  ##add explanation for all the options!!!!!!!
+"""
+
+import functools
+import json
+import os
+
+from object_detection import evaluator
+from object_detection import trainer
+from object_detection.builders import dataset_builder
+from object_detection.builders import graph_rewriter_builder
+from object_detection.builders import model_builder
+from object_detection.utils import config_util
+from object_detection.utils import dataset_util
+from object_detection.utils import label_map_util
+
+import tensorflow as tf
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+flags = tf.app.flags
+flags.DEFINE_string('master', '', 'Name of the TensorFlow master to use.')
+flags.DEFINE_integer('task', 0, 'task id')
+flags.DEFINE_integer('num_clones', 1, 'Number of clones to deploy per worker.')
+flags.DEFINE_boolean('clone_on_cpu', False,
+                     'Force clones to be deployed on CPU.  Note that even if '
+                     'set to False (allowing ops to run on gpu), some ops may '
+                     'still be run on the CPU if they have no GPU kernel.')
+flags.DEFINE_integer('worker_replicas', 1, 'Number of worker+trainer '
+                                           'replicas.')
+flags.DEFINE_integer('ps_tasks', 0,
+                     'Number of parameter server tasks. If None, does not use '
+                     'a parameter server.')
+flags.DEFINE_string('train_dir', '',
+                    'Directory to save the checkpoints and training summaries.')
+
+flags.DEFINE_string('pipeline_config_path', '',
+                    'Path to a pipeline_pb2.TrainEvalPipelineConfig config '
+                    'file. If provided, other configs are ignored')
+
+flags.DEFINE_boolean('eval_training_data', False,
+                     'If training data should be evaluated for this job.')
+# check point dir is the same as the train dir
+# flags.DEFINE_string('checkpoint_dir', '',
+#                    'Directory containing checkpoints to evaluate, typically '
+#                    'set to `train_dir` used in the training job.')
+flags.DEFINE_string('eval_dir', '',
+                    'Directory to write eval summaries to.')
+
+flags.DEFINE_boolean('run_once', False, 'Option to only run a single pass of '
+                                        'evaluation. Overrides the `max_evals`'
+                                        ' parameter in the provided config.')
+flags.DEFINE_float('box_min_ap', 1, 'Option to run until the box average'
+                                    'precision reaches this number')
+flags.DEFINE_float('mask_min_ap', 1, 'Option to run until the mask average'
+                                     'precision reaches this number')
+FLAGS = flags.FLAGS
+
+
+def train(_):
+  # TODO(mehdi): training 1 epoch
+  pass
+
+
+def evaluate(_):
+  # TODO(mehdi): evaluating with mask and box metrics
+  pass
+
+
+def main(_):
+  assert FLAGS.train_dir, '`train_dir` is missing.'
```

Comment:
You can set the flag default to `None` and use `mark_flag_as_required to mark required flags. For example:
```
flags.DEFINE_string('train_dir', None, '...')
flags.mark_flag_as_required('train_dir')
```

Reply:
why? does it work as the "assert FLAGS. ..."